glue/plugins/scatter_group/client.py

```diff
@@ -0,0 +1,239 @@
+
+from .util import get_colors
+from ...clients.scatter_client import ScatterClient
+from ...clients.layer_artist import (ChangedTrigger, ScatterLayerArtist)
+
+
+class ScatterGroupClient(ScatterClient):
+
+    """
+    A client class that uses matplotlib to visualize tables as scatter plots.
+    """
+    xmin = CallbackProperty(0)
+    xmax = CallbackProperty(1)
+    ymin = CallbackProperty(0)
+    ymax = CallbackProperty(1)
+    ylog = CallbackProperty(False)
+    xlog = CallbackProperty(False)
+    yflip = CallbackProperty(False)
+    xflip = CallbackProperty(False)
+    xatt = CallbackProperty()
+    yatt = CallbackProperty()
+    gatt = CallbackProperty()
+    jitter = CallbackProperty()
+
+    def _connect(self):
+        add_callback(self, 'xlog', self._set_xlog)
+        add_callback(self, 'ylog', self._set_ylog)
+        add_callback(self, 'xflip', self._set_limits)
+        add_callback(self, 'yflip', self._set_limits)
+        add_callback(self, 'xmin', self._set_limits)
+        add_callback(self, 'xmax', self._set_limits)
+        add_callback(self, 'ymin', self._set_limits)
+        add_callback(self, 'ymax', self._set_limits)
+        add_callback(self, 'xatt', partial(self._set_xydata, 'x'))
+        add_callback(self, 'yatt', partial(self._set_xydata, 'y'))
+        add_callback(self, 'gatt', partial(self._set_xydata, 'g'))
+        add_callback(self, 'jitter', self._jitter)
+        self.axes.figure.canvas.mpl_connect('draw_event',
+                                            lambda x: self._pull_properties())
+
+    def groupable_attributes(self, layer, show_hidden=False):
+        data = layer.data
+        l = data._shape[0]
+        if not data.find_component_id('None'):
+            none_comp = Component(np.array(range(0, l)), units='None')
+            data.add_component(none_comp, 'None', hidden=False)
+        else:
+            none_comp = data.find_component_id('None')
+            to_comp = coerce_numeric(np.array(range(0, l)))
+            to_comp.setflags(write=False)
+            none_comp._data = to_comp
+
+        comp = data.components if show_hidden else data.visible_components
+        groups = [comp[-1]]
+        for c in comp:
+            if data.get_component(c).group:
+                groups.append(c)
+        return groups
+
+    def add_layer(self, layer):
+        """ Adds a new visual layer to a client, to display either a dataset
+        or a subset. Updates both the client data structure and the
+        plot.
+
+        Returns the created layer artist
+
+        :param layer: the layer to add
+        :type layer: :class:`~glue.core.data.Data` or :class:`~glue.core.subset.Subset`
+        """
+        if layer.data not in self.data:
+            raise TypeError("Layer not in data collection")
+        if layer in self.artists:
+            return self.artists[layer][0]
+
+        result = ScatterGroupLayerArtist(layer, self.axes)
```

Comment:
Hmm this is the only changed line in this function, which suggests theres a better way to organize the original class. What about changing the main ScatterClient.add_layer to something like `result = self.layer_artist_class(layer, self.axes)`, setting `layer_artist_class=ScatterLayerArtist` at the class level in the main class, and `layer_artist_class=ScatterGroupLayerArtist` in your class?